connectors/vmware/src/centreon/script/centreon_vmware.pm

```diff
@@ -145,29 +145,98 @@ sub new {
+
+    $self->{logger}->writeLogDebug("Reading configuration from " . $self->{opt_extra});
+    my $centreon_vmware_config_from_json;
+
+    if ($self->{opt_extra} =~ /.*\.pm$/i) {
+        our %centreon_vmware_config;
+        # loads the .pm configuration (compile time)
+        require($self->{opt_extra});
+        # Concatenation of the default parameters with the ones from the config file
+        $self->{centreon_vmware_config} = {%{$self->{centreon_vmware_default_config}}, %centreon_vmware_config};
+    } elsif ($self->{opt_extra} =~ /.*\.json$/i) {
+        $centreon_vmware_config_from_json = $self->parse_json_file();
+        # The structure of the JSON is different from the .pm file. The code was designed to work with the latter, so
+        # the structure of $self->{centreon_vmware_config} must be adapted after parsing to avoid a massive refactoring of
+        # the whole program. The wanted structure is a key-object dictionary instead of an array of objects.
+        my %vsphere_server_dict = map { lc($_->{name}) => $_ } @{$centreon_vmware_config_from_json->{vsphere_server}};
+        # Replace the "raw" structure from the JSON file.
+        $centreon_vmware_config_from_json->{vsphere_server} = \%vsphere_server_dict;
+        # Concatenation of the default parameters with the ones from the config file
+        $self->{centreon_vmware_config} = {%{$self->{centreon_vmware_default_config}}, %$centreon_vmware_config_from_json};
+    }
+}
+
+# report_config_check: writes a report of what has been loaded from the configuration file
+sub report_config_check {
+    my ($self, %options) = @_;
+
+    my $nb_server_entries = scalar(keys %{$self->{centreon_vmware_config}->{vsphere_server}});
+    my $entry_spelling    = ($nb_server_entries > 1) ? 'entries' : 'entry';
+
+    my $report = "Configuration file " . $self->{opt_extra} . " has been read correctly and has ";
+    $report .=         $nb_server_entries . " " . $entry_spelling . ".";
+
+    $self->{logger}->writeLogInfo($report);
+    return $report;
+}
+
 sub init {
     my $self = shift;
     $self->SUPER::init();
 
     # redefine to avoid out when we try modules
     $SIG{__DIE__} = undef;
 
-    if (!defined($self->{opt_extra})) {
+    if ( ! defined($self->{opt_extra}) ) {
         $self->{opt_extra} = "/etc/centreon/centreon_vmware.pm";
     }
-    if (-f $self->{opt_extra}) {
-        require $self->{opt_extra};
-    } else {
-        $self->{logger}->writeLogInfo("Can't find extra config file $self->{opt_extra}");
-    }
 
-    $self->{centreon_vmware_config} = {%{$self->{centreon_vmware_default_config}}, %centreon_vmware_config};
+    if ( ! -f $self->{opt_extra} ) {
+        my $msg = "Can't find config file '$self->{opt_extra}'. If a migration from "
+            . "/etc/centreon/centreon_vmware.pm to /etc/centreon/centreon_vmware.json is required, you may "
+            . "run perl convert_config_file.pl /etc/centreon/centreon_vmware.pm > /etc/centreon/centreon_vmware.json";
```

Comment:
please Add the full path of convert_config_file in the message.
Also I don't see it in this pr
maybe you should make it executable and without file extention.